src/Interpreters/executeQuery.cpp

```diff
@@ -346,9 +350,38 @@ QueryLogElement logQueryStart(
 
         if (elem.type >= settings.log_queries_min_type && !settings.log_queries_min_query_duration_ms.totalMilliseconds())
         {
-            if (auto query_log = context->getQueryLog())
-                query_log->add(elem);
+            if (!settings.log_query_settings && settings.log_query_settings.changed)
+                LOG_DEBUG(
+                    getLogger("executeQuery"),
+                    "Query settings will not be added to query_log since setting `log_query_settings` has been set to false."
+                    " The setting has been changed for the query");
+
+            query_log->add(elem);
+        }
+        else if (elem.type < settings.log_queries_min_type)
+        {
+            if (settings.log_queries_min_type.changed)
```

Comment:
Could we combine the outer and inner ifs (l. 361/363) to a single if, similar to l. 353?

(same for below two nested ifs)